pkg/util/pki/parse.go

```diff
@@ -336,3 +339,7 @@ func (c *chainNode) root() *chainNode {
 
 	return c
 }
+
+func isSelfSignedCertificate(cert *x509.Certificate) bool {
+	return cert.Issuer.CommonName == cert.Subject.CommonName
```

Comment:
This is not a strong enough check for whether the certificate is self-signed:

```suggestion
	return cert.CheckSignatureFrom(cert) == nil
```